Fixed ltur-notifier with current LTUR webpage

mloeks opened the pull request:
Hi priska,

I stumpled upon your little script after browsing the LTUR webpage many times, wondering if there isn't a better and automatic way to do this. Also, I keep forgetting about the low fares for my weekend travels and frequently pay to much... ;-)

Great work, many thanks for sharing. I was glad I found your repo.

I had a look on your code and the most recent webpage of LTUR and was able to fix the script.
The key was, that the form used for entering the journey details is dynamically loaded via AJAX and not part of the initial page.

I had to adjust the scraping algorithm. It now uses BeautifulSoup selections with only some slight changes to what is being searched for (LTUR changed the tag names/attributes).

Don't be confused by the many lines changed in commit 5d4527c, that was just because of some IDE indentation reformatting. The crucial changes can be seen in commit 5c67ac4.

It would be great if you had a brief look on my changes and merged my fork to your master branch if you like to.
Let me know if you have any questions!

Cheers,
Matthias

Commit:
reformatted code because IDE complained
added comment on what's failing currently and ideas for possible solutions

Commit:
fixed price scraper with latest version of LTUR webpage
found out URL of form which is now dynamically loaded
replaced finding algorithm with BeautifulSoup based selection
added pip requirements.txt file

Commit:
updated README

Commit:
prepared new dir structure for fundamental revision and enhancement of
the code

policecar commented:
Hey Matthias,

thanks for updating the code! I have been wanting to do that.

It's also neat seeing that other people find the snippet useful.

I have some comments /questions regarding the pull request: It seems to me that the files src/ltur_models.py and test/test_ltur_models.py aren't really used. If that is so, would you mind removing them and the pytest dependency?

I'm also getting an ImportError for the config when running ltur_notifier.py because the module name is missing. Do you mind adapting that, too, or should I?

Cheers,
priska

Commit:
reverted last commit. the latest changes were not meant to go into the forked repo.
might create my own copy or branch for future developments.
this status should be merged back to the origin/master if agreed by the original developer.

matze09 commented:
Hi Priska,

I'm sorry, the latest commit 22bb330 was not meant for this fork. I just prepared something for future development, which I might do in an own branch or copied repo.

I reverted everything to commit 9342886. This is the status, which should be merged back into your repo, if you agree with the changes. 

Sorry about the confusion, but I'm not that experienced with GIT. :-)

Please test this version and let me know if you have any questions or if anything doesn't work as expected.

Cheers,
Matthias

policecar commented:
Thanks for your PR, Matthias!
